File: Book My Show/ClientApp/src/app/Models/Enum.ts

    export enum Language
    {
        Other = 0,
        Hindi,   
        Engilsh,
        Tamil,
        Telgu,
    }

    export const LanguageLabel = new Map<number,string>([
        [Language.Other,"Other"],
        [Language.Hindi,"Hindi"],
        [Language.Engilsh,"Engilsh"],
        [Language.Tamil,"Tamil"],
        [Language.Telgu,"Telgu"]
    ]);


    export enum Gender
    {
        Other = 0,        
        Male ,
        Female ,
    }

    export enum Role
    {
        Other,
        Actor,
        Actress,
        SupportingActor,
        SupportingActress,        
        Producer,
        Director,
    }
    export const RoleLabel = new Map<Role,string>([
        [Role.Other,"Other"],
        [Role.Actor,"Actor"],
        [Role.Actress,"Actress"],
        [Role.SupportingActor,"Supporting Actor"],
        [Role.SupportingActress,"Supporting Actress"],
        [Role.Producer,"Producer"],
        [Role.Director,"Director"],
    ]);

    

    export enum Gener
    {   
        Other,   
        Action,
        Comedy,
        Drama,
        Fantasy,
        Horror,
        Mystery,
        Romance,
        Thriller
    }

    export const GenresLabel = new Map<Gener,string>([
        [Gener.Other,"Other"],
        [Gener.Action,"Action"],
        [Gener.Comedy,"Comedy"],
        [Gener.Drama,"Drama"],
        [Gener.Fantasy,"Fantasy"],
        [Gener.Horror,"Horror"],
        [Gener.Mystery,"Mystery"],
        [Gener.Romance,"Romance"],
        [Gener.Thriller,"Thriller"],
    ]);

    export enum MovieType
    {
        Other,
        _2D,
        _3D,
        IMAX,
        IMAX3D,
    }

    export const MovieTypeLabel = new Map<number,string>([
        [MovieType.Other,"Other"],
        [MovieType._2D,"2D"],
        [MovieType._3D,"3D"],
        [MovieType.IMAX,"IMAX"],
        [MovieType.IMAX3D,"IMAX 3d"]
    ]);

    export enum CertificateType
    {
        Other,
        U,
        UA,
        A,
    }

    export enum SeatType
    {
        Other,
        FrontRow,
        MiddleRow,
        LastRow,
    }

File: Book My Show/Models/Enum.cs
using System;
using System.ComponentModel;
using System.Collections.Generic;
using System.Text;

namespace Models.Enums
{
    public enum Language
    {
        [Description("Other")]
        Other = 0,
        [Description("Hindi")]
        Hindi,
        [Description("English")]
        Engilsh,
        [Description("Tamil")]
        Tamil,
        [Description("Telgu")]
        Telgu,
    }

    public enum Gender
    {
        [Description("Other")]
        Other = 0,
        [Description("Male")]
        Male = 1,
        [Description("Female")]
        Female = 2,
    }

    public enum Role
    {

        [Description("Other")]
        Other,
        [Description("Actor")]
        Actor,
        [Description("Actress")]
        Actress,
        [Description("Supporting Actor")]
        SupportingActor,
        [Description("Supporting Actress")]
        SupportingActress,
        [Description("Producer")]
        Producer,
        [Description("Director")]
        Director,
    }

    public enum Gener
    {
        [Description("Other")]
        Other,
        [Description("Action")]
        Action,
        [Description("Comedy")]
        Comedy,
        [Description("Drama")]
        Drama,
        [Description("Fantasy")]
        Fantasy,
        [Description("Horror")]
        Horror,
        [Description("Mystery")]
        Mystery,
        [Description("Romance")]
        Romance,
        [Description("Thriller")]
        Thriller
    }

    public enum MovieType
    {
        [Description("Other")]
        Other,
        [Description("2D")]
        _2D,
        [Description("3D")]
        _3D,
        [Description("IMAX")]
        IMAX,
        [Description("IMAX 3D")]
        IMAX3D,
    }

    public enum CertificateType
    {
        [Description("Other")]
        Other,
        [Description("All Age Groups")]
        U,
        [Description("Above 12 yrs")]
        UA,
        [Description("Above 18 yrs")]
        A,
    }

    public enum SeatType
    {
        [Description("Other")]
        Other,
        [Description("First Row Seat")]
        FrontRow,
        [Description("Middle Row Seat")]
        MiddleRow,
        [Description("Last Row Seat")]
        LastRow,
    }

}


// Access the Description of the Enum
//public string SlotType
//{
//    get
//    {
//        return MyExtensions.GetDescription(this.TypeOfSlot);
//    }
//}